From 8037fecaf72a59835d66585b19ca09c4df12c09a Mon Sep 17 00:00:00 2001
From: Alex Suykov <alex.suykov@gmail.com>
Date: Wed, 31 Dec 2014 11:21:44 +0200
Subject: [PATCH 5/7] arm: syscall()

---
 libcompat/syscall.S | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/libcompat/syscall.S b/libcompat/syscall.S
index 3884329..c0a7b3e 100644
--- a/libcompat/syscall.S
+++ b/libcompat/syscall.S
@@ -143,6 +143,18 @@ syscall:
 	la	$25, __unified_syscall
 	jr	$25
 
+#elif defined(__arm__) && defined __ARM_EABI__
+	stmfd	sp!,{r4,r5,r7,lr}
+	mov	r7, r0
+	mov	r0, r1
+	mov	r1, r2
+	mov	r2, r3
+	ldr	r3, [sp,#16]
+	ldr	r4, [sp,#20]
+	ldr	r5, [sp,#24]
+	swi	0
+	b	__unified_syscall
+
 #else
 #include <endian.h>
 	.section	.comment
-- 
2.0.3

